storage/grpc_writer.go

```diff
@@ -453,16 +451,15 @@ type gRPCResumableBidiWriteBufferSender struct {
 	stream            storagepb.Storage_BidiWriteObjectClient
 	flushOffset       int64
 	settings          *settings
+	writer            *gRPCWriter
```

Comment:
I think we'd probably want to give a callback for storage.Writer to set the checksum rather than piping this through as a field.

Also, we should update the docs on storage.Writer to clarify the new contract for providing a checksum.